### PR TITLE
fix: dont log "couldn't connect to miner" in 2.5

### LINKED ISSUE #4757: Remove signer log when DKG doesn't succeed
When a new 2.5 reward cycle starts and the aggregate key isn't set, the signer has this log every few seconds:
 
 ```
Signer #N: Could not lookup current miner while in active reward cycle
```

I thought we'd removed this log during 2.5, but looks like we haven't.

Relates to:

- #4729

### PR DESCRIPTION
**Pull request by hstove**
- Closes https://github.com/stacks-network/stacks-core/issues/4757

When the signer looks up the current coordinator during an active reward cycle, it assumes the miner should be the coordinator. In 2.5, this isn't the case. When DKG doesn't complete before the active reward cycle, the signer continues to lookup the coordinator, but it still outputs this error log.

TLDR: the signer's coordinator logic is fine, as it already falls back to the DKG coordinator, but it shouldn't log this error in 2.5.

To implement this, I've extended the `RewardCycleInfo` struct to include the current epoch, and I pass around this struct instead of just `current_reward_cycle` in a few places.

### CODE REVIEW

**Review by obycode — APPROVED** [review]
LGTM!

File: stacks-signer/src/runloop.rs [hunk 1]
```diff
@@ -427,7 +430,7 @@ impl SignerRunLoop<Vec<OperationResult>, RunLoopCommand> for RunLoop {
                 &self.stacks_client,
                 event.as_ref(),
                 res.clone(),
-                current_reward_cycle,
+                &current_reward_cycle_info,
```
Comment (jferrant) [inline stacks-signer/src/runloop.rs:433]: Don't think this needs to be by reference here.

File: stacks-signer/src/runloop.rs [hunk 2]
```diff
@@ -450,7 +453,7 @@ impl SignerRunLoop<Vec<OperationResult>, RunLoopCommand> for RunLoop {
                 }
             }
             // After processing event, run the next command for each signer
-            signer.process_next_command(&self.stacks_client, current_reward_cycle);
+            signer.process_next_command(&self.stacks_client, &current_reward_cycle_info);
```
Comment (jferrant) [inline stacks-signer/src/runloop.rs:456]: similarly here. Don't think you need the reference.

File: stacks-signer/src/signer.rs [hunk 3]
```diff
@@ -516,7 +522,7 @@ impl Signer {
                     // We cannot execute a DKG command if we are not the coordinator
                     Some(self.get_coordinator_dkg().0)
                 } else {
-                    self.get_coordinator_sign(current_reward_cycle).0
+                    self.get_coordinator_sign(&current_reward_cycle_info).0
```
Comment (jferrant) [inline stacks-signer/src/signer.rs:525]: Simiarly don't htink you need the reference here.

File: stacks-signer/src/signer.rs [hunk 4]
```diff
@@ -653,13 +659,18 @@ impl Signer {
                     let coordinator_pubkey = if Self::is_dkg_message(&packet.msg) {
                         self.get_coordinator_dkg().1
                     } else {
-                        self.get_coordinator_sign(current_reward_cycle).1
+                        self.get_coordinator_sign(&current_reward_cycle_info).1
```
Comment (jferrant) [inline stacks-signer/src/signer.rs:662]: Same here.

**Comment by jferrant** [comment]
A bit unrelated to this PR, but it would be nice if you updated the internal call to get_node_epoch in process_dkg to go away (this was done in a sep branch that also added epoch to reward cycle info, but that branch is dead). Just need to pass the reward cycle info down to it. Its not a blocker for this PR, so if you don't want to do it here, will make a ticket and do on top of this.

**Comment by hstove** [comment]
I think this should be closed given the v0/v1 changes, yeah? If not, I think it'd be best to rebase on top of #4778 . @jferrant what do you think?

**Comment by jferrant** [comment]
> I think this should be closed given the v0/v1 changes, yeah? If not, I think it'd be best to rebase on top of #4778 . @jferrant what do you think?

Haha I am useless. yes that made sense. But now you can rebase on top of develop as v0 is merged!

**Comment by hstove** [comment]
Closing this - no longer relevant
